[Regression] [Beta] : Cannot build apk on Flutter 1.25.0-8.1.pre (applies to example apps for Flutter plugins too)

I maintain a plugin that builds on the the stable channel (1.22.5) but had gotten a report that it fails to build on the current beta version (1.25.0-8.1.pre) [here](https://github.com/MaikuB/flutter_local_notifications/issues/950). Was able to reproduce the reported issue by building the example app (https://github.com/MaikuB/flutter_local_notifications/tree/master/flutter_local_notifications/example) whilst on the beta channel.

Furthermore, if I clone https://github.com/flutter/plugins and run the `flutter build apk` command on the `url_launcher` plugin I get similar issues where the output is as follows

<details>
<summary>url_launcher build logs</summary>

```
FAILURE: Build failed with an exception.                                
                                                                        
* What went wrong:                                                      
Could not determine the dependencies of task ':integration_test:compileReleaseAidl'.
> Could not resolve all task dependencies for configuration ':integration_test:releaseCompileClasspath'.
   > Could not resolve androidx.annotation:annotation:{strictly 1.0.0}. 
     Required by:                                                       
         project :integration_test                                      
      > Cannot find a version of 'androidx.annotation:annotation' that satisfies the version constraints: 
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.test:runner:1.2.0' --> 'androidx.annotation:annotation:1.0.0'
           Constraint path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.annotation:annotation:{strictly 1.0.0}' because of the following reason: releaseRuntimeClasspath uses version 1.0.0
           Constraint path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.annotation:annotation:{strictly 1.0.0}' because of the following reason: releaseRuntimeClasspath uses version 1.0.0
           Constraint path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.annotation:annotation:{strictly 1.0.0}' because of the following reason: releaseRuntimeClasspath uses version 1.0.0
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.test:runner:1.2.0' --> 'androidx.test:monitor:1.2.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-common:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-common-java8:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-runtime:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-runtime:2.2.0' --> 'androidx.arch.core:core-common:2.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.collection:collection:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.viewpager:viewpager:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.loader:loader:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.activity:activity:1.0.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.core:core:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.core:core:1.1.0' --> 'androidx.versionedparcelable:versionedparcelable:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.viewpager:viewpager:1.0.0' --> 'androidx.customview:customview:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.lifecycle:lifecycle-viewmodel:2.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.activity:activity:1.0.0' --> 'androidx.savedstate:savedstate:1.0.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.loader:loader:1.0.0' --> 'androidx.lifecycle:lifecycle-livedata:2.0.0' --> 'androidx.arch.core:core-runtime:2.0.0' --> 'androidx.annotation:annotation:1.0.0'
                                                                        
   > Could not resolve androidx.annotation:annotation:1.0.0.            
     Required by:                                                       
         project :integration_test > androidx.test:runner:1.2.0         
         project :integration_test > androidx.test:monitor:1.2.0        
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.viewpager:viewpager:1.0.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.loader:loader:1.0.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.viewpager:viewpager:1.0.0 > androidx.customview:customview:1.0.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.loader:loader:1.0.0 > androidx.lifecycle:lifecycle-livedata:2.0.0 > androidx.arch.core:core-runtime:2.0.0
      > Cannot find a version of 'androidx.annotation:annotation' that satisfies the version constraints: 
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.test:runner:1.2.0' --> 'androidx.annotation:annotation:1.0.0'
           Constraint path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.annotation:annotation:{strictly 1.0.0}' because of the following reason: releaseRuntimeClasspath uses version 1.0.0
           Constraint path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.annotation:annotation:{strictly 1.0.0}' because of the following reason: releaseRuntimeClasspath uses version 1.0.0
           Constraint path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.annotation:annotation:{strictly 1.0.0}' because of the following reason: releaseRuntimeClasspath uses version 1.0.0
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.test:runner:1.2.0' --> 'androidx.test:monitor:1.2.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-common:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-common-java8:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-runtime:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-runtime:2.2.0' --> 'androidx.arch.core:core-common:2.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.collection:collection:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.viewpager:viewpager:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.loader:loader:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.activity:activity:1.0.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.core:core:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.core:core:1.1.0' --> 'androidx.versionedparcelable:versionedparcelable:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.viewpager:viewpager:1.0.0' --> 'androidx.customview:customview:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.lifecycle:lifecycle-viewmodel:2.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.activity:activity:1.0.0' --> 'androidx.savedstate:savedstate:1.0.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.loader:loader:1.0.0' --> 'androidx.lifecycle:lifecycle-livedata:2.0.0' --> 'androidx.arch.core:core-runtime:2.0.0' --> 'androidx.annotation:annotation:1.0.0'
                                                                        
   > Could not resolve androidx.annotation:annotation:1.1.0.            
     Required by:                                                       
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.lifecycle:lifecycle-common:2.2.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.lifecycle:lifecycle-common-java8:2.2.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.lifecycle:lifecycle-runtime:2.2.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.lifecycle:lifecycle-runtime:2.2.0 > androidx.arch.core:core-common:2.1.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.core:core:1.1.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.collection:collection:1.1.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.activity:activity:1.0.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.core:core:1.1.0 > androidx.versionedparcelable:versionedparcelable:1.1.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.lifecycle:lifecycle-viewmodel:2.1.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.activity:activity:1.0.0 > androidx.savedstate:savedstate:1.0.0
      > Cannot find a version of 'androidx.annotation:annotation' that satisfies the version constraints: 
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.test:runner:1.2.0' --> 'androidx.annotation:annotation:1.0.0'
           Constraint path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.annotation:annotation:{strictly 1.0.0}' because of the following reason: releaseRuntimeClasspath uses version 1.0.0
           Constraint path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.annotation:annotation:{strictly 1.0.0}' because of the following reason: releaseRuntimeClasspath uses version 1.0.0
           Constraint path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.annotation:annotation:{strictly 1.0.0}' because of the following reason: releaseRuntimeClasspath uses version 1.0.0
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.test:runner:1.2.0' --> 'androidx.test:monitor:1.2.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-common:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-common-java8:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-runtime:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-runtime:2.2.0' --> 'androidx.arch.core:core-common:2.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.collection:collection:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.viewpager:viewpager:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.loader:loader:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.activity:activity:1.0.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.core:core:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.core:core:1.1.0' --> 'androidx.versionedparcelable:versionedparcelable:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.viewpager:viewpager:1.0.0' --> 'androidx.customview:customview:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.lifecycle:lifecycle-viewmodel:2.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.activity:activity:1.0.0' --> 'androidx.savedstate:savedstate:1.0.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.loader:loader:1.0.0' --> 'androidx.lifecycle:lifecycle-livedata:2.0.0' --> 'androidx.arch.core:core-runtime:2.0.0' --> 'androidx.annotation:annotation:1.0.0'
                                                                        
                                                                        
* Try:                                                                  
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
                                                                        
* Get more help at https://help.gradle.org                              
                                                                        
BUILD FAILED in 2m 12s
```
</details>

Had similar issues building the example app of the `google_maps_flutter` plugin as well

<details>
<summary>google_maps_flutter build logs</summary>

```
FAILURE: Build failed with an exception.                                
                                                                        
* What went wrong:                                                      
Could not determine the dependencies of task ':integration_test:compileReleaseAidl'.
> Could not resolve all task dependencies for configuration ':integration_test:releaseCompileClasspath'.
   > Could not resolve androidx.annotation:annotation:{strictly 1.0.0}. 
     Required by:                                                       
         project :integration_test                                      
      > Cannot find a version of 'androidx.annotation:annotation' that satisfies the version constraints: 
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.test:runner:1.2.0' --> 'androidx.annotation:annotation:1.0.0'
           Constraint path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.annotation:annotation:{strictly 1.0.0}' because of the following reason: releaseRuntimeClasspath uses version 1.0.0
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.test:runner:1.2.0' --> 'androidx.test:monitor:1.2.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-common:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-common-java8:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-runtime:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-runtime:2.2.0' --> 'androidx.arch.core:core-common:2.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.collection:collection:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.viewpager:viewpager:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.loader:loader:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.activity:activity:1.0.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.core:core:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.viewpager:viewpager:1.0.0' --> 'androidx.customview:customview:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.lifecycle:lifecycle-viewmodel:2.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.activity:activity:1.0.0' --> 'androidx.savedstate:savedstate:1.0.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.loader:loader:1.0.0' --> 'androidx.lifecycle:lifecycle-livedata:2.0.0' --> 'androidx.arch.core:core-runtime:2.0.0' --> 'androidx.annotation:annotation:1.0.0'
                                                                        
   > Could not resolve androidx.annotation:annotation:1.0.0.            
     Required by:                                                       
         project :integration_test > androidx.test:runner:1.2.0         
         project :integration_test > androidx.test:monitor:1.2.0        
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.viewpager:viewpager:1.0.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.loader:loader:1.0.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.viewpager:viewpager:1.0.0 > androidx.customview:customview:1.0.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.loader:loader:1.0.0 > androidx.lifecycle:lifecycle-livedata:2.0.0 > androidx.arch.core:core-runtime:2.0.0
      > Cannot find a version of 'androidx.annotation:annotation' that satisfies the version constraints: 
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.test:runner:1.2.0' --> 'androidx.annotation:annotation:1.0.0'
           Constraint path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.annotation:annotation:{strictly 1.0.0}' because of the following reason: releaseRuntimeClasspath uses version 1.0.0
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.test:runner:1.2.0' --> 'androidx.test:monitor:1.2.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-common:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-common-java8:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-runtime:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-runtime:2.2.0' --> 'androidx.arch.core:core-common:2.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.collection:collection:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.viewpager:viewpager:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.loader:loader:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.activity:activity:1.0.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.core:core:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.viewpager:viewpager:1.0.0' --> 'androidx.customview:customview:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.lifecycle:lifecycle-viewmodel:2.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.activity:activity:1.0.0' --> 'androidx.savedstate:savedstate:1.0.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.loader:loader:1.0.0' --> 'androidx.lifecycle:lifecycle-livedata:2.0.0' --> 'androidx.arch.core:core-runtime:2.0.0' --> 'androidx.annotation:annotation:1.0.0'
                                                                        
   > Could not resolve androidx.annotation:annotation:1.1.0.            
     Required by:                                                       
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.lifecycle:lifecycle-common:2.2.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.lifecycle:lifecycle-common-java8:2.2.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.lifecycle:lifecycle-runtime:2.2.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.lifecycle:lifecycle-runtime:2.2.0 > androidx.arch.core:core-common:2.1.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.core:core:1.1.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.collection:collection:1.1.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.activity:activity:1.0.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.lifecycle:lifecycle-viewmodel:2.1.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.activity:activity:1.0.0 > androidx.savedstate:savedstate:1.0.0
      > Cannot find a version of 'androidx.annotation:annotation' that satisfies the version constraints: 
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.test:runner:1.2.0' --> 'androidx.annotation:annotation:1.0.0'
           Constraint path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.annotation:annotation:{strictly 1.0.0}' because of the following reason: releaseRuntimeClasspath uses version 1.0.0
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.test:runner:1.2.0' --> 'androidx.test:monitor:1.2.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-common:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-common-java8:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-runtime:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-runtime:2.2.0' --> 'androidx.arch.core:core-common:2.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.collection:collection:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.viewpager:viewpager:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.loader:loader:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.activity:activity:1.0.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.core:core:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.viewpager:viewpager:1.0.0' --> 'androidx.customview:customview:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.lifecycle:lifecycle-viewmodel:2.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.activity:activity:1.0.0' --> 'androidx.savedstate:savedstate:1.0.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.loader:loader:1.0.0' --> 'androidx.lifecycle:lifecycle-livedata:2.0.0' --> 'androidx.arch.core:core-runtime:2.0.0' --> 'androidx.annotation:annotation:1.0.0'
                                                                        
                                                                        
* Try:                                                                  
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
                                                                        
* Get more help at https://help.gradle.org                              
                                                                        
BUILD FAILED in 15s   
```
                      
</details>

Included details from running `flutter doctor -v`
<details>
<summary>flutter doctor -v output</summary>

```
[✓] Flutter (Channel beta, 1.25.0-8.1.pre, on macOS 11.0.1 20B50 darwin-x64,
    locale en-AU)
    • Flutter version 1.25.0-8.1.pre at /Users/michaelbui/flutter
    • Framework revision 8f89f6505b (3 days ago), 2020-12-15 15:07:52 -0800
    • Engine revision 92ae191c17
    • Dart version 2.12.0 (build 2.12.0-133.2.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/michaelbui/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.17.0

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 87.0.4280.88
```
</details>

I am able to replicate this issue on latest beta (1.25) wherein trying to run `flutter build apk` for even a sample app is returning the error as above mentioned.  The same error is being shown while running `flutter/plugins` examples as well (tried with `url_launcher`, `google_maps_flutter` and `image_picker`)

<details>
<summary>compile log</summary>

```
FAILURE: Build failed with an exception.                                
                                                                        
* What went wrong:                                                      
Could not determine the dependencies of task ':integration_test:compileReleaseAidl'.
> Could not resolve all task dependencies for configuration ':integration_test:releaseCompileClasspath'.
   > Could not resolve androidx.annotation:annotation:{strictly 1.0.0}. 
     Required by:                                                       
         project :integration_test                                      
      > Cannot find a version of 'androidx.annotation:annotation' that satisfies the version constraints: 
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.test:runner:1.2.0' --> 'androidx.annotation:annotation:1.0.0'
           Constraint path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.annotation:annotation:{strictly 1.0.0}' because of the following reason: releaseRuntimeClasspath uses version 1.0.0
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.test:runner:1.2.0' --> 'androidx.test:monitor:1.2.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-common:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-common-java8:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-runtime:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-runtime:2.2.0' --> 'androidx.arch.core:core-common:2.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.collection:collection:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.viewpager:viewpager:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.loader:loader:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.activity:activity:1.0.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.core:core:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.viewpager:viewpager:1.0.0' --> 'androidx.customview:customview:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.lifecycle:lifecycle-viewmodel:2.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.activity:activity:1.0.0' --> 'androidx.savedstate:savedstate:1.0.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.loader:loader:1.0.0' --> 'androidx.lifecycle:lifecycle-livedata:2.0.0' --> 'androidx.arch.core:core-runtime:2.0.0' --> 'androidx.annotation:annotation:1.0.0'
                                                                        
   > Could not resolve androidx.annotation:annotation:1.0.0.            
     Required by:                                                       
         project :integration_test > androidx.test:runner:1.2.0         
         project :integration_test > androidx.test:monitor:1.2.0        
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.viewpager:viewpager:1.0.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.loader:loader:1.0.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.viewpager:viewpager:1.0.0 > androidx.customview:customview:1.0.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.loader:loader:1.0.0 > androidx.lifecycle:lifecycle-livedata:2.0.0 > androidx.arch.core:core-runtime:2.0.0
      > Cannot find a version of 'androidx.annotation:annotation' that satisfies the version constraints: 
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.test:runner:1.2.0' --> 'androidx.annotation:annotation:1.0.0'
           Constraint path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.annotation:annotation:{strictly 1.0.0}' because of the following reason: releaseRuntimeClasspath uses version 1.0.0
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.test:runner:1.2.0' --> 'androidx.test:monitor:1.2.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-common:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-common-java8:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-runtime:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-runtime:2.2.0' --> 'androidx.arch.core:core-common:2.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.collection:collection:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.viewpager:viewpager:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.loader:loader:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.activity:activity:1.0.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.core:core:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.viewpager:viewpager:1.0.0' --> 'androidx.customview:customview:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.lifecycle:lifecycle-viewmodel:2.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.activity:activity:1.0.0' --> 'androidx.savedstate:savedstate:1.0.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.loader:loader:1.0.0' --> 'androidx.lifecycle:lifecycle-livedata:2.0.0' --> 'androidx.arch.core:core-runtime:2.0.0' --> 'androidx.annotation:annotation:1.0.0'
                                                                        
   > Could not resolve androidx.annotation:annotation:1.1.0.            
     Required by:                                                       
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.lifecycle:lifecycle-common:2.2.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.lifecycle:lifecycle-common-java8:2.2.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.lifecycle:lifecycle-runtime:2.2.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.lifecycle:lifecycle-runtime:2.2.0 > androidx.arch.core:core-common:2.1.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.core:core:1.1.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.collection:collection:1.1.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.activity:activity:1.0.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.lifecycle:lifecycle-viewmodel:2.1.0
         project :integration_test > io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3 > androidx.fragment:fragment:1.1.0 > androidx.activity:activity:1.0.0 > androidx.savedstate:savedstate:1.0.0
      > Cannot find a version of 'androidx.annotation:annotation' that satisfies the version constraints: 
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.test:runner:1.2.0' --> 'androidx.annotation:annotation:1.0.0'
           Constraint path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.annotation:annotation:{strictly 1.0.0}' because of the following reason: releaseRuntimeClasspath uses version 1.0.0
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'androidx.test:runner:1.2.0' --> 'androidx.test:monitor:1.2.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-common:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-common-java8:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-runtime:2.2.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.lifecycle:lifecycle-runtime:2.2.0' --> 'androidx.arch.core:core-common:2.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.collection:collection:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.viewpager:viewpager:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.loader:loader:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.activity:activity:1.0.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.core:core:1.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.viewpager:viewpager:1.0.0' --> 'androidx.customview:customview:1.0.0' --> 'androidx.annotation:annotation:1.0.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.lifecycle:lifecycle-viewmodel:2.1.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.activity:activity:1.0.0' --> 'androidx.savedstate:savedstate:1.0.0' --> 'androidx.annotation:annotation:1.1.0'
           Dependency path 'com.example.integration_test:integration_test:1.0-SNAPSHOT' --> 'io.flutter:flutter_embedding_release:1.0.0-92ae191c17a53144bf4d62f3863c110be08e3fd3' --> 'androidx.fragment:fragment:1.1.0' --> 'androidx.loader:loader:1.0.0' --> 'androidx.lifecycle:lifecycle-livedata:2.0.0' --> 'androidx.arch.core:core-runtime:2.0.0' --> 'androidx.annotation:annotation:1.0.0'
                                                                        
                                                                        
* Try:                                                                  
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
                                                                        
* Get more help at https://help.gradle.org                              
                                                                        
BUILD FAILED in 1m 28s                                                  
Running Gradle task 'assembleRelease'...                                
Running Gradle task 'assembleRelease'... Done                      89.1s
The build failed likely due to AndroidX incompatibilities in a plugin. The tool is about to try using Jetifier to solve the incompatibility.
✏️  Creating `android/settings_aar.gradle`...                        2ms
✓ `android/settings_aar.gradle` created successfully.
Building plugin flutter_plugin_android_lifecycle...
Running Gradle task 'assembleAarRelease'...                             
Running Gradle task 'assembleAarRelease'... Done                    6.6s


FAILURE: Build failed with an exception.

* What went wrong:
Task 'assembleAarRelease' not found in root project 'flutter_plugin_android_lifecycle'.

* Try:
Run gradlew tasks to get a list of available tasks. Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 5s


The plugin flutter_plugin_android_lifecycle could not be built due to the issue above.


```

</details>

<details>
<summary>latest beta flutter doctor -v</summary>

```
[✓] Flutter (Channel beta, 1.25.0-8.1.pre, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-IN)
    • Flutter version 1.25.0-8.1.pre at /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 8f89f6505b (5 days ago), 2020-12-15 15:07:52 -0800
    • Engine revision 92ae191c17
    • Dart version 2.12.0 (build 2.12.0-133.2.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.51.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 87.0.4280.88

• No issues found!


```

</details>

Tried the same on latest master (1.26) which compiled and built apk without any issues.

<details>
<summary>latest master compile log</summary>

```
dhs@Dhss-MacBook-Pro example % flutter build apk    
Downloading android-arm-profile/darwin-x64 tools...                585ms
Downloading android-arm-release/darwin-x64 tools...                260ms
Downloading android-arm64-profile/darwin-x64 tools...              314ms
Downloading android-arm64-release/darwin-x64 tools...              499ms
Downloading android-x64-profile/darwin-x64 tools...                732ms
Downloading android-x64-release/darwin-x64 tools...                612ms
Running "flutter pub get" in example...                            685ms
You are building a fat APK that includes binaries for android-arm, android-arm64, android-x64.
If you are deploying the app to the Play Store, it's recommended to use app bundles or split the APK to reduce the APK size.
    To generate an app bundle, run:
        flutter build appbundle --target-platform android-arm,android-arm64,android-x64
        Learn more: https://developer.android.com/guide/app-bundle
    To split the APKs per ABI, run:
        flutter build apk --target-platform android-arm,android-arm64,android-x64 --split-per-abi
        Learn more: https://developer.android.com/studio/build/configure-apk-splits#configure-abi-split
Calling mockable JAR artifact transform to create file: /Users/dhs/.gradle/caches/transforms-2/files-2.1/996752eba439264efa9dfd3d70eef921/android.jar with input /Users/dhs/Library/Android/sdk/platforms/android-29/android.jar
Removed unused resources: Binary resource data reduced from 45KB to 36KB: Removed 20%
Running Gradle task 'assembleRelease'...                                
Running Gradle task 'assembleRelease'... Done                     125.8s
✓ Built build/app/outputs/flutter-apk/app-release.apk (16.2MB).


```

</details>

<details>
<summary>latest master flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 1.26.0-2.0.pre.86, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-IN)
    • Flutter version 1.26.0-2.0.pre.86 at
      /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision b793285329 (3 days ago), 2020-12-18 10:14:36 -0800
    • Engine revision 7983c6fda0
    • Dart version 2.12.0 (build 2.12.0-167.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.51.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • Mac OS X 10.15.4 19E2269
      darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 87.0.4280.88

! Doctor found issues in 1 category.


```

</details>

Although the same code sample and `flutter/plugins` examples are compiling properly on previous beta version (1.23).  Tried by running `url_launcher` plugin example:

<details>
<summary>previous beta compile log</summary>

```

dhs@Dhss-MacBook-Pro example % flutter build apk
Running "flutter pub get" in example...                             2.4s
You are building a fat APK that includes binaries for android-arm, android-arm64, android-x64.
If you are deploying the app to the Play Store, it's recommended to use app bundles or split the APK to reduce the APK size.
    To generate an app bundle, run:
        flutter build appbundle --target-platform android-arm,android-arm64,android-x64
        Learn more: https://developer.android.com/guide/app-bundle
    To split the APKs per ABI, run:
        flutter build apk --target-platform android-arm,android-arm64,android-x64 --split-per-abi
        Learn more: https://developer.android.com/studio/build/configure-apk-splits#configure-abi-split
Removed unused resources: Binary resource data reduced from 45KB to 36KB: Removed 20%
Running Gradle task 'assembleRelease'...                                
Running Gradle task 'assembleRelease'... Done                      64.8s
✓ Built build/app/outputs/flutter-apk/app-release.apk (16.1MB).

```

</details>

<details>
<summary>1.23 beta flutter doctor -v</summary>

```
[✓] Flutter (Channel beta, 1.23.0-18.1.pre, on Mac OS X 10.15.4 19E2269 x86_64,
    locale en-IN)
    • Flutter version 1.23.0-18.1.pre at /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 198df796aa (10 weeks ago), 2020-10-15 12:04:33 -0700
    • Engine revision 1d12d82d9c
    • Dart version 2.11.0 (build 2.11.0-213.1.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.51.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (2 available)
    • Web Server (web) • web-server • web-javascript • Flutter Tools
    • Chrome (web)     • chrome     • web-javascript • Google Chrome
      87.0.4280.88

• No issues found!



```

</details>

--- a/dev/devicelab/bin/tasks/android_plugin_example_app_build_test.dart
+++ b/dev/devicelab/bin/tasks/android_plugin_example_app_build_test.dart
@@ -15,7 +15,6 @@ final String gradlewExecutable = Platform.isWindows ? '.\\$gradlew' : './$gradle
 /// Tests that a plugin example app can be built using the current Flutter Gradle plugin.
 Future<void> main() async {
   await task(() async {
-
     section('Find Java');
 
     final String javaHome = await findJavaHome();
@@ -32,22 +31,47 @@ Future<void> main() async {
       options: <String>['--android', '--no-ios'],
     );
 
-    final Directory tempDir = Directory.systemTemp.createTempSync('flutter_plugin_test.');
-    final Directory projectDir = Directory(path.join(tempDir.path, 'plugin_test'));
+    final Directory tempDir =
+        Directory.systemTemp.createTempSync('flutter_plugin_test.');
+    final Directory projectDir =
+        Directory(path.join(tempDir.path, 'plugin_test'));
     try {
       await inDirectory(tempDir, () async {
         await flutter(
           'create',
-          options: <String>['--template=plugin', '--platforms=android', 'plugin_test'],
+          options: <String>[
+            '--template=plugin',
+            '--platforms=android',
+            'plugin_test',
+          ],
         );
       });
 
-      final Directory exampleAppDir = Directory(path.join(projectDir.path, 'example'));
+      final Directory exampleAppDir =
+          Directory(path.join(projectDir.path, 'example'));
       if (!exists(exampleAppDir)) {
         return TaskResult.failure('Example app directory doesn\'t exist');
       }
 
-      section('Run flutter build apk');
+      final File buildGradleFile =
+          File(path.join(exampleAppDir.path, 'android', 'build.gradle'));
+
+      if (!exists(buildGradleFile)) {
+        return TaskResult.failure('$buildGradleFile doesn\'t exist');
+      }
+
+      final String buildGradle = buildGradleFile.readAsStringSync();
+      final RegExp androidPluginRegExp =
+          RegExp(r'com\.android\.tools\.build:gradle:(\d+\.\d+\.\d+)');
+
+      section('Use AGP 4.1.0');
+
+      String newBuildGradle = buildGradle.replaceAll(
+          androidPluginRegExp, 'com.android.tools.build:gradle:4.1.0');
+      print(newBuildGradle);
+      buildGradleFile.writeAsString(newBuildGradle);
+
+      section('Run flutter build apk using AGP 4.1.0');
 
       await inDirectory(exampleAppDir, () async {
         await flutter(
@@ -67,6 +91,55 @@ Future<void> main() async {
         'flutter-apk',
         'app-release.apk',
       );
+
+      if (!exists(File(exampleApk))) {
+        return TaskResult.failure('Failed to build app-release.apk');
+      }
+
+      section('Clean');
+
+      await inDirectory(exampleAppDir, () async {
+        await flutter('clean');
+      });
+
+      section('Remove Gradle wrapper');
+
+      Directory(path.join(exampleAppDir.path, 'android', 'gradle', 'wrapper'))
+          .deleteSync(recursive: true);
+
+      section('Use AGP 3.3.0');
+
+      newBuildGradle = buildGradle.replaceAll(
+          androidPluginRegExp, 'com.android.tools.build:gradle:3.3.0');
+      print(newBuildGradle);
+      buildGradleFile.writeAsString(newBuildGradle);
+
+      section('Enable R8 in gradle.properties');
+
+      final File gradleProperties =
+          File(path.join(exampleAppDir.path, 'android', 'gradle.properties'));
+
+      if (!exists(gradleProperties)) {
+        return TaskResult.failure('$gradleProperties doesn\'t exist');
+      }
+
+      gradleProperties.writeAsString('''
+org.gradle.jvmargs=-Xmx1536M
+android.useAndroidX=true
+android.enableJetifier=true
+android.enableR8=true''');
+
+      section('Run flutter build apk using AGP 3.3.0');
+
+      await inDirectory(exampleAppDir, () async {
+        await flutter(
+          'build',
+          options: <String>[
+            'apk',
+            '--target-platform=android-arm',
+          ],
+        );
+      });
 
       if (!exists(File(exampleApk))) {
         return TaskResult.failure('Failed to build app-release.apk');

--- a/packages/flutter_tools/gradle/flutter.gradle
+++ b/packages/flutter_tools/gradle/flutter.gradle
@@ -20,6 +20,7 @@ import org.gradle.api.tasks.InputFiles
 import org.gradle.api.tasks.OutputDirectory
 import org.gradle.api.tasks.TaskAction
 import org.gradle.api.tasks.bundling.Jar
+import org.gradle.util.VersionNumber
 
 buildscript {
     repositories {
@@ -304,7 +305,7 @@ class FlutterPlugin implements Plugin<Project> {
         project.dependencies {
             implementation pluginProject
         }
-        Closure addEmbeddingCompileOnlyDependency = { buildType ->
+        Closure addEmbeddingDependencyToPlugin = { buildType ->
             String flutterBuildMode = buildModeFor(buildType)
             // In AGP 3.5, the embedding must be added as an API implementation,
             // so java8 features are desugared against the runtime classpath.
@@ -317,15 +318,37 @@ class FlutterPlugin implements Plugin<Project> {
             pluginProject.android.buildTypes {
                 "${buildType.name}" {}
             }
-            pluginProject.dependencies.add(
-                "${buildType.name}CompileOnly",
-                "io.flutter:flutter_embedding_$flutterBuildMode:$engineVersion")
+            // The embedding is a compileOnly dependency of a Flutter plugin,
+            // however prior to Gradle 6.0.0, it must be an API dependency.
+            //
+            // Not doing so, causes transitive dependency resolution conflicts.
+            // That is, the embedding dependencies resolved in the plugin are
+            // different than the ones resolved in the app.
+            if (isGradleVersionGraterOrEqualThan('6.0.0')) {
+              addCompileOnlyDependency(
+                pluginProject,
+                buildType.name,
+                "io.flutter:flutter_embedding_$flutterBuildMode:$engineVersion"
+              )
+            } else {
+              addApiDependencies(
+                pluginProject,
+                buildType.name,
+                "io.flutter:flutter_embedding_$flutterBuildMode:$engineVersion"
+              )
+            }
         }
         // Wait until the Android plugin loaded.
         pluginProject.afterEvaluate {
-            project.android.buildTypes.each addEmbeddingCompileOnlyDependency
-            project.android.buildTypes.whenObjectAdded addEmbeddingCompileOnlyDependency
+            project.android.buildTypes.each addEmbeddingDependencyToPlugin
+            project.android.buildTypes.whenObjectAdded addEmbeddingDependencyToPlugin
         }
+    }
+
+    // Returns `true` if the current Gradle version is greater or equal to the given version.
+    private isGradleVersionGraterOrEqualThan(String version) {
+      return VersionNumber.parse(project.gradle.gradleVersion)
+          .compareTo(VersionNumber.parse(version)) >= 0
     }
 
     // Returns `true` if the given path contains an `android/build.gradle` file.

